### DISCUSSION TITLE
Android version

### DISCUSSION THREAD

**Discussion by axredneck** [Orama-Interactive/Pixelorama#68] [Ideas]
Is it possible to port Pixelorama to Android?

**Comment by Calinou** [comment]
This is likely feasible from a technical standpoint, but the UI isn't designed to be used on a small screen. It will also likely drain the device's battery until Godot's low-processor mode is implemented on mobile platforms.

**Comment by axredneck** [comment]
Screen is not small if it's a tablet.

**Comment by Calinou** [comment]
@axredneck In this case, it still has to be designed to be used with a touchscreen rather than a keyboard and mouse :slightly_smiling_face:

This implies the need for larger buttons, among other things.

**Comment by OverloadedOrama** [comment]
I actually did some Android experiments yesterday. The UI is indeed very small, but I think this can be fixed. Not sure how but we would eventually figure it out. The real issue is the FileDialog nodes. No matter how much I've tried, I can't open and/or save files in Android.

**Comment by AndreevAndrei** [comment]
And it need to support a stylus (stick) for Samsung Galaxy Nodes 🤣 
Or morden iPad sticks)

**Comment by axredneck** [comment]
@OverloadedOrama does it have a permission to save files?

**Comment by OverloadedOrama** [comment]
> @OverloadedOrama does it have a permission to save files?

It's been a while since I last tested it, but last time I checked I did give permission to save and load files. I think they're the "Read External Storage" and "Write External Storage" permissions found in Godot's Android export settings.

**Reply by Variable-ind** [reply]
Hy, i just stumbled upon this discussion, to save something  we also need to use `OS.request_permissions()` at-leat once in code to enable those permissions. I made a "Diary" app using Godot that lets the user save his diary on phone storage in form of .txt files.
The source code of the app is available [(Here!)](https://github.com/Variable-ind/My_Diary) hope it helps.

**Reply by SergReminder** [reply]
Yes, it helps

**Comment by ikeofkc** [comment]
Just note that all chromebooks now work with android apps, and you can publish in a way like krita's highly adored android port, that only allows it to work on those devices.

**Comment by ArmainAP** [comment]
You can simply run Pixelorama from browser via the itch.io
I am viewing this on Lenovo Yoga Tab 13 and the UI is use able without any issues. Stylus works too except for the buttons.
